File: android/app/src/main/java/jagtheora/ogg/OggPage.java
package jagtheora.ogg;

import jagtheora.misc.SimplePeer;

public final class OggPage extends SimplePeer {

    public long getGranulePos() { return 0L; }

    public boolean isBOS() { return false; }

    public int getVersion() { return 0; }

    public int getCompletedPackets() { return 0; }

    public int getSerialNumber() { return 0; }

    public boolean isEOS() { return false; }

    protected void clear() {}

    public boolean isContinued() { return false; }

    public long getPageNumber() { return 0L; }
}

File: src/jvmMain/kotlin/jagtheora/ogg/OggPage.kt
package jagtheora.ogg

import jagtheora.misc.SimplePeer

actual class OggPage : SimplePeer() {
    /*private*/ actual val version: Int
        external get

    /*private*/ actual val isContinued: Boolean
        external get

    actual val serialNumber: Int
        external get

    /*private*/ actual val isEOS: Boolean
        external get

    /*private*/ actual val granulePos: Long
        external get

    actual val isBOS: Boolean
        external get

    actual external override fun clear()

    /*private*/ actual val pageNumber: Long
        external get

    /*private*/ actual val completedPackets: Int
        external get
}
